### PR TITLE
doxygen updates to module_write_internal_state.F90 and io/module_write_netcdf.F90

### PR DESCRIPTION
**Pull request by edwardhartnett**
## Description

doxygen updates to io/module_write_netcdf.F90m and module_write_netcdf.F90.

We need some information filled in by @DusanJovic-NOAA and @junwang-noaa (look for "???" and replace it with documentation).

### Issue(s) addressed

Part of #760 

## Testing

Only documentation changes in this PR.

## Dependencies

N/A

### CODE REVIEW

**Comment by edwardhartnett** [comment]
Thanks @DusanJovic-NOAA !

This PR is ready to review and merge.

File: io/module_write_internal_state.F90 [hunk 1]
```diff
@@ -1,16 +1,10 @@
...
-!       Feb 2017:  J. Wang - Initial code
-!
-!-----------------------------------------------------------------------
-!
+!> @file
+!> @brief The internal state of the write component.
+!> @author Jun Wang @date Feb, 2017
+
+!> @brief The internal state of the write component.
+!> @author Jun Wang @date Feb, 2017
```
Comment (DusanJovic-NOAA) [inline io/module_write_internal_state.F90:6]: Why is this repeated twice?

Comment (edwardhartnett) [reply io/module_write_internal_state.F90:6]: Once for the file, once for the module. Doxygen needs both.

File: io/module_write_internal_state.F90 [hunk 2]
```diff
@@ -20,12 +14,27 @@ module write_internal_state
 !-----------------------------------------------------------------------
 !
       type output_grid_info
-        integer :: im, jm, lm
-        integer :: i_start,i_end, j_start,j_end
-        real,dimension(:,:),allocatable  :: lonPtr, latPtr
-        integer,dimension(:),allocatable :: i_start_wrtgrp, i_end_wrtgrp, j_start_wrtgrp, j_end_wrtgrp
-        real    :: latse, latnw, lonse, lonnw
-        real    :: latstart, latlast, lonstart, lonlast
+        integer :: im !< output grid global I dimension size
```
Comment (edwardhartnett) [inline io/module_write_internal_state.F90]: @junwang-noaa and @DusanJovic-NOAA No big deal for these, but in future, please capitalize and put a period at the end. For example:

` integer :: im !< Output grid global I dimension size.`

It makes the final documentation look a lot better.

File: io/module_write_netcdf.F90 [hunk 3]
```diff
@@ -19,11 +28,21 @@ module module_write_netcdf
...
+
+  !> Write netCDF file.
+  !>
+  !> @param[in] wrtfb ESMF write field bundle
+  !> @param[in] filename NetCDF filename.
+  !> @param[in] use_parallel_netcdf True if parallel I/O should be used.
+  !> @param[in] mpi_comm MPI communicator for parallel I/O.
+  !> @param[in] mype MPI rank
+  !> @param[in] grid_id Output grid identifier
+  !> @param[out] rc Return code: return code
```
Comment (edwardhartnett) [inline io/module_write_netcdf.F90]: @DusanJovic-NOAA what is the rc for successful completion of the function? 0?

It seems that sometimes the rc is an ESMF error code, and other times a more local error code. Ideally we would like all the possible return codes here, but if not, then at least let's say what code is returned for success.

Comment (DusanJovic-NOAA) [reply io/module_write_netcdf.F90]: Ok. Thanks.

File: io/module_write_internal_state.F90 (hunk 1 above)
Comment (DusanJovic-NOAA) [reply io/module_write_internal_state.F90:6]: Ok. Thanks.

File: io/module_write_netcdf.F90 (hunk 3 above)
Comment (DusanJovic-NOAA) [reply io/module_write_netcdf.F90]: rc is equal to ESMF_SUCCESS if there are no errors. The value of ESMF_SUCCESS is 0.

Comment (DusanJovic-NOAA) [reply io/module_write_netcdf.F90]: The list of all possible return codes is long.

https://earthsystemmodeling.org/docs/release/latest/ESMF_refdoc/node9.html#SECTION09030000000000000000

Comment (edwardhartnett) [reply io/module_write_netcdf.F90]: OK, I will change it to:

return code - 0 for success, ESMF error code otherwise.

Comment (edwardhartnett) [reply io/module_write_netcdf.F90]: OK, I have made that change.

**Comment by edwardhartnett** [comment]
This PR is ready to merge...

**Comment by junwang-noaa** [comment]
@DusanJovic-NOAA would you please test this PR in UFS before committing it? Thanks

**Comment by DusanJovic-NOAA** [comment]
@edwardhartnett Please sync this branch with develop.

**Comment by edwardhartnett** [comment]
@DusanJovic-NOAA I have synced.

**Comment by DusanJovic-NOAA** [comment]
> @DusanJovic-NOAA I have synced.

This branch (ejh_d5) is still one commit behind.

**Comment by edwardhartnett** [comment]
@DusanJovic-NOAA sorry, wrong branch! I've updated ejh_d5 now and we should be good to go...

**Comment by DusanJovic-NOAA** [comment]
Test passed on Hera. [RegressionTests_hera.log](https://github.com/NOAA-EMC/fv3atm/files/14044146/RegressionTests_hera.log)

Ready to merge.

**Review by junwang-noaa — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**